File: Level2/기능개발.py
def solution(progress,speeds):
    answer=[]
    #걸리는 일수
    days=[]
    
    #남은 일수로 총 몇 일걸리는지 
    for i in range(len(progress)):
        rest_progress = 100 - progress[i]
        if rest_progress % speeds[i]==0:
            days.append(rest_progress // speeds[i])
        else:
            days.append((rest_progress // speeds[i])+1)
            
    #비교
    cmp=days[0]
    day=1
    for i in range(1,len(days)):
        #배포가 불가능하면
        if cmp>=days[i]:
            day+=1
        
        #가능하면
        else:
            answer.append(day)
            cmp=days[i]
            day=1
            
    #마지막
    answer.append(day) 
    
    #반환
    return answer

File: Level2/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    int size = progresses.size();
    vector<int> answer, works;
    
    //남은 작업량
    for (int i = 0; i < size; ++i) {
        int p = 100 - progresses[i];
        if ((p % speeds[i]) == 0)
            works.push_back(p / speeds[i]);
        else
            works.push_back((p / speeds[i]) + 1);
    }
    
    //비교 및 카운트
    int cur_work = works[0], cnt = 1;
    
    for (int i = 1; i <size; ++i) {
        if (cur_work >= works[i])  //현재 작업량이 끝났다면
            cnt++;
        
        else { //교체
            answer.push_back(cnt);
            cur_work = works[i];
            cnt = 1;
        }
    }
    answer.push_back(cnt);
    
    return answer;
}
